### PR TITLE
Break down flow.ControlBehavior to flow.TokenCalculateStrategy and flow.ControlBehavior

### PR DESCRIPTION
**Pull request by louyuting**
### Describe what this PR does / why we need it
Break down flow.ControlBehavior to flow.AllocationStrategy and flow.ControlEffect
```go
const (
	Default AllocationStrategy = iota
	WarmUp
)

const (
	Reject ControlEffect = iota
	Throttling
)
```
### Special notes for reviews

### CODE REVIEW

File: core/flow/rule.go [hunk 1]
```diff
@@ -34,16 +45,62 @@ const (
...
+	default:
+		return "Undefined"
+	}
+}
+
+type AllocationStrategy int32
 
 const (
-	Reject ControlBehavior = iota
+	Default AllocationStrategy = iota
```
Comment (sczyh30) [inline core/flow/rule.go:62]: "Default" seems too generic?

Comment (louyuting) [reply core/flow/rule.go:62]: Maybe Direct?

Comment (louyuting) [reply core/flow/rule.go]: The naming had been updated. 
How do you think of?

File: core/flow/rule.go [hunk 2]
```diff
@@ -15,55 +15,112 @@ const (
...
+}
+
 // RelationStrategy indicates the flow control strategy based on the relation of invocations.
 type RelationStrategy int32
 
 const (
-	// Direct means flow control by current resource directly.
-	Direct RelationStrategy = iota
+	// DirectResource means flow control by current resource directly.
+	DirectResource RelationStrategy = iota
```
Comment (sczyh30) [inline core/flow/rule.go:34]: DirectResource seems confusing. Maybe "CurrentResource`?

Comment (louyuting) [reply core/flow/rule.go:34]: Make sense

**Review by sczyh30 — APPROVED** [review]

**Comment by sczyh30** [comment]
Thanks!

**Pull request merged**